internal/config/mount_config.go

```diff
@@ -77,6 +78,11 @@ type GrpcClientConfig struct {
 	ConnPoolSize int `yaml:"conn-pool-size,omitempty"`
 }
 
+type BucketFlow struct {
```

Comment:
I am not sure if bucket-flow is easy to understand for end users. Should we rename it to something like "gcs-client: enable-hns", you can take check with Marco & Kislay on this.

Reply:
@ankitaluthra1 suggested `BucketLevelFeatureToggle`.

Reply:
After discussions with Marco and Kislay, we've decided to keep the feature flag as `enable-hns`. Marco had initially left the decision to the engineering team, as this feature won't be public-facing. Following further discussion with Kislay, we concluded that enable-hns is the best choice, since the likelihood of additional bucket-level features is low.